src/averageAges.js

```diff
@@ -55,7 +65,24 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const findChildren = ({ mother }) => {
+    return people.some(({ name }) => mother === name);
+  };
+  const findMother = ({ mother }) => {
+    return people.find(({ name }) => name === mother);
+  };
+
+  let children = Array.from(people);
+
+  (onlyWithSon) && (children = children.filter(({ sex }) => sex === 'm'));
+  children = children.filter(findChildren);
+
+  const women = children.map(findMother);
```

Comment:
this looks strange
let's use && and other operators inside array methods, not outside of them
as one variant of solving this is too create an array of age differences and push there ages of the children found depending where the onlyWithSon flag is passed or not
you can do it with forEach and map combined